common/src/main/java/org/opensearch/sdk/SdkClient.java

```diff
@@ -80,7 +85,51 @@ default GetDataObjectResponse getDataObject(GetDataObjectRequest request) {
             if (cause instanceof InterruptedException) {
                 Thread.currentThread().interrupt();
             }
-            throw new OpenSearchException(cause);
+            // Rethrow unchecked Exceptions
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            } else {
+                throw new OpenSearchException(cause);                
+            }
+        }
+    }
+
+    /**
+     * Update a data object/document in a table/index.
+     * @param request A request identifying the data object to update
+     * @param executor the executor to use for asynchronous execution
+     * @return A completion stage encapsulating the response or exception
+     */
+    public CompletionStage<UpdateDataObjectResponse> updateDataObjectAsync(UpdateDataObjectRequest request, Executor executor);
+
+    /**
+     * Update a data object/document in a table/index.
+     * @param request A request identifying the data object to update
+     * @return A completion stage encapsulating the response or exception
+     */
+    default CompletionStage<UpdateDataObjectResponse> updateDataObjectAsync(UpdateDataObjectRequest request) {
+        return updateDataObjectAsync(request, ForkJoinPool.commonPool());        
+    }
+
+    /**
+     * Update a data object/document in a table/index.
+     * @param request A request identifying the data object to update
+     * @return A response on success. Throws {@link OpenSearchException} wrapping the cause on exception.
+     */
+    default UpdateDataObjectResponse updateDataObject(UpdateDataObjectRequest request) {
+        try {
+            return updateDataObjectAsync(request).toCompletableFuture().join();
+        } catch (CompletionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            // Rethrow unchecked Exceptions
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            } else {
+                throw new OpenSearchException(cause);                
+            }
```

Comment:
if all these are same for every methods, may be we can put that in a method to reuse?

Reply:
`ExceptionsHelper.convertToRuntime()` would be perfect for this but takes an Exception as an argument... guess I have to copy it. 🤷